File: ProjectEuler/Problems/Problem003.cs
using Microsoft.Extensions.Logging;
using System.Linq;

namespace ProjectEuler.Problems
{
    public class Problem003 : Problem<long>
    {
        public Problem003(ILogger<Problem003> logger) : base(logger) { }

        protected override long CalculateAnswer()
        {
            var answer = 0L;
            var n = 600851475143L;
            var primeFactorization = MathHelpers.PrimeFactorization(n);

            Logger.LogDebug("Prime Factors:");
            foreach (var prime in primeFactorization.Entries.Select(x => x.Prime))
            {
                Logger.LogDebug("{PrimeFactor}", prime);
                if (prime > answer) answer = prime;
            }

            return answer;
        }
    }
}

File: ProjectEulerPython/problems/problem_003.py
import math_helpers
from problems.problem import Problem

class Problem003(Problem):

  def calculate_answer(self) -> int:
    answer = 0
    n = 600851475143
    prime_factorization = math_helpers.prime_factorization(n)

    self.print_detail("Prime factors:")
    for entry in prime_factorization.entries():
      self.print_detail(entry.prime)
      if (entry.prime) > answer:
        answer = entry.prime

    return answer